### PR TITLE
docs(manager/dockerfile): Document supported dependencies for Dockerfile manager

### LINKED ISSUE #29872: Explain `ARG IMAGE=` updates in `dockerfile` manager docs
### Describe the proposed change(s).

Todos:

- [ ] Read the code snippets from the maintainer: https://github.com/renovatebot/renovate/discussions/28916#discussioncomment-9349722
- [ ] Improve the [Renovate docs, Dockerfile manager](https://docs.renovatebot.com/modules/manager/dockerfile/) so it explains how you can get Renovate to update the `ARG IMAGE=` parts (use the pencil icon on the published docs page to get to the correct file)

I don't understand the code enough to write the docs improvements, so I'll let someone else pick up this issue. 😉

### PR DESCRIPTION
**Pull request by malmor**
## Changes

This pull requests adds a `Supported dependencies` section to the `Dockerfile` manager documentation.

## Context

The `Dockerfile` manager supports many different types of dependencies out of the box. This pull requests adds them to the documentation. I tested the different types of updates (`FROM`, `COPY --from`, `syntax`) and ensured they actually work as expected.

This pull request also fixes a small code highlighting issue in the `Flux` manager documentation (`'` was being used instead of `` ` `` for two of the resources).

Closes #29872 

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/flux/readme.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 This manager parses [Flux](https://fluxcd.io/) YAML manifests and supports:
 
 1. [`HelmRelease`](https://fluxcd.io/docs/components/helm/helmreleases/) resources
-1. ['GitRepository'](https://fluxcd.io/flux/components/source/gitrepositories/) resources
-1. ['OCIRepository'](https://fluxcd.io/flux/components/source/ocirepositories/) resources
-1. Flux [system](https://fluxcd.io/docs/installation) manifests
+2. [`GitRepository`](https://fluxcd.io/flux/components/source/gitrepositories/) resources
+3. [`OCIRepository`](https://fluxcd.io/flux/components/source/ocirepositories/) resources
+4. Flux [system](https://fluxcd.io/docs/installation) manifests
```
Comment (rarkins) [inline lib/modules/manager/flux/readme.md:6]: We keep ordered list unnumbered intentionally - please revert this change

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/modules/manager/dockerfile/readme.md [hunk 2]
```diff
@@ -1,4 +1,51 @@
-Extracts image references in a `Dockerfile` and/or `Containerfile`.
+### Supported dependencies
+
+This manager extracts image references in a `Dockerfile` and/or `Containerfile` and supports:
+
+1. [`FROM`](https://docs.docker.com/reference/dockerfile/#from) images
```
Comment (HonkingGoose) [inline lib/modules/manager/dockerfile/readme.md:5]:
```suggestion
- [`FROM`](https://docs.docker.com/reference/dockerfile/#from) images
```

File: lib/modules/manager/dockerfile/readme.md [hunk 3]
```diff
@@ -1,4 +1,51 @@
...
+1. [`COPY --from`](https://docs.docker.com/reference/dockerfile/#copy---from) images
```
Comment (HonkingGoose) [inline lib/modules/manager/dockerfile/readme.md:6]:
```suggestion
- [`COPY --from`](https://docs.docker.com/reference/dockerfile/#copy---from) images
```

File: lib/modules/manager/dockerfile/readme.md [hunk 4]
```diff
@@ -1,4 +1,51 @@
...
+1. [`syntax`](https://docs.docker.com/reference/dockerfile/#syntax) images
```
Comment (HonkingGoose) [inline lib/modules/manager/dockerfile/readme.md:7]:
```suggestion
- [`syntax`](https://docs.docker.com/reference/dockerfile/#syntax) images
```

File: lib/modules/manager/dockerfile/readme.md [hunk 5]
```diff
@@ -1,4 +1,51 @@
...
+
+#### `FROM` support
+
+Renovate can update images referenced in `FROM` directives. This also works for multi stage builds where multiple `FROM` directives are being used in a single `Dockerfile`.
```
Comment (HonkingGoose) [inline lib/modules/manager/dockerfile/readme.md:11]:
```suggestion
Renovate can update images referenced in `FROM` directives.
This even works for multi-stage builds with many `FROM` directives in one Dockerfile.
```

File: lib/modules/manager/dockerfile/readme.md [hunk 6]
```diff
@@ -1,4 +1,51 @@
...
+FROM node:20.9.0
+```
+
+Advanced `FROM` flags like `--platform` or `AS <name>` are also supported:
+
+```dockerfile
+FROM --platform=linux/amd64 node:20.9.0 AS installer
+```
+
+Also variables and [`ARG` directives](https://docs.docker.com/reference/dockerfile/#understand-how-arg-and-from-interact) will automatically be expanded:
```
Comment (HonkingGoose) [inline lib/modules/manager/dockerfile/readme.md:23]:
```suggestion
Also, Renovate will automatically expand variables and [`ARG` directives](https://docs.docker.com/reference/dockerfile/#understand-how-arg-and-from-interact):
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
Changes:

- Use a plain list.
- One sentence per line.
- Rewrite to shorten sentences, and use simpler words.

Good catch on the bad `code` styling in the other docs page!

**Review by rarkins — APPROVED** [review]

**Pull request merged**